File: Leetcode_500.py

class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        row1 = "qwertyuiop"
        row2 = "asdfghjkl"
        row3 = "zxcvbnm"
        result = []

        for word in words:
            rows = [0, 0, 0]
            for ch in word.lower():
                if ch in row1:
                    rows[0] = 1
                elif ch in row2:
                    rows[1] = 1
                elif ch in row3:
                    rows[2] = 1

            if sum(rows) == 1:
                result.append(word)

        return result

File: Leetcode_500.java
class Solution {
    public List<String> findWords(String[] words) {
        String row1 = "qwertyuiop";
        String row2 = "asdfghjkl";
        String row3 = "zxcvbnm";
        
        List<String> result = new ArrayList<>();
        
        for (String word : words) {
            int[] rows = {0, 0, 0};
            String lowerWord = word.toLowerCase();
            
            for (char ch : lowerWord.toCharArray()) {
                if (row1.indexOf(ch) != -1) {
                    rows[0] = 1;
                } else if (row2.indexOf(ch) != -1) {
                    rows[1] = 1;
                } else if (row3.indexOf(ch) != -1) {
                    rows[2] = 1;
                }
            }

            if (rows[0] + rows[1] + rows[2] == 1) {
                result.add(word); // Add original case word
            }
        }
        
        return result;
    }
}
